trino/trino.go

```diff
@@ -1943,3 +2012,19 @@ func (s *NullSlice3Map) Scan(value interface{}) error {
 	s.Valid = true
 	return nil
 }
+
+type QueryProgressCallbackResponse struct {
+	ID         string    `json:"id"`
+	QueryStats stmtStats `json:"stmtStats"`
+	Error      stmtError `json:"error"`
+}
+
+type queryProgressCallbackPeriod struct {
+	Period           time.Duration `json:"period"`
+	LastCallbackTime time.Time     `json:"lastCallbackTime"`
+}
+
+type QueryProgressCallback interface {
```

Comment:
It would be idiomatic in Go to call this `ProgressUpdater` and have a method called `Update`.